assets/approval.py

```diff
@@ -0,0 +1,344 @@
+
+            else:
+                log.info("The lock does not exist")
+                exit(1)
+        else:
+            log.error('Please use an available action')
+            exit(1)
+
+        metadata = []
+        for key in approval_lock.keys_():
+            value = getattr(approval_lock, key)
+            if type(value) is datetime:
+                value = str(Decimal(value.timestamp()))
+            if type(value) is bool:
+                value = str(value)
+
+            metadata.append(
+                {
+                    'name': key,
+                    'value': value
+                }
+            )
+
+        name_path = os.path.join(target_dir, 'name')
+        with open(name_path, 'w') as name:
+            name.write(approval_lock.lockname)
+
+        metadata_path = os.path.join(target_dir, 'metadata')
+        with open(metadata_path, 'w') as metadata_file:
+            json.dump(metadata, metadata_file)
+
+        return {
+            'version': {"timestamp": "{timestamp}".format(
+                timestamp=Decimal(approval_lock.timestamp.timestamp()))},
+            'metadata': metadata,
+        }
+
+    def run(self):
+        """Parse input/arguments, perform requested command return output."""
+        # Extract informations from the json
+        source = self.data.get('source', {})
+        params = self.data.get('params', {})
+        version = self.data.get('version', {})
+
+        # To use AWS with efficiency, we are pushing the AWS credentials into environment
+        os.environ['AWS_ACCESS_KEY_ID'] = source.get('AWS_ACCESS_KEY_ID', '')
+        os.environ['AWS_SECRET_ACCESS_KEY'] = source.get('AWS_SECRET_ACCESS_KEY', '')
+        os.environ['AWS_DEFAULT_REGION'] = source.get('AWS_DEFAULT_REGION', "eu-west-1")
+        self.wait_lock = source.get('wait_lock', 10)
+
+        # Ensure we are receiving the required parameters on the configuration
+        if 'pool' not in source:
+            log.error("pool must exist in the source configuration")
+            exit(1)
+        else:
+            self.pool = source.get('pool')
+
+        # Configure the connection to Dynamodb
+        self.engine.connect_to_region(os.environ.get('AWS_DEFAULT_REGION', 'eu-west-1'))
+        # Register our model with the engine so it can create the Dynamo table
+        self.engine.register(Approval)
+        # Create the dynamo table for our registered model
+        self.engine.create_schema()
+
+        # Define which operation to perform
+        if self.command_name == 'check':
+            response = self.check_cmd(source, version)
+        elif self.command_name == 'in':
+            response = self.in_cmd(self.command_argument[0], source, version, params)
+        else:
+            response = self.out_cmd(self.command_argument[0], source, params)
+
+        return json.dumps(response)
+
+print(ApprovalResource(os.path.basename(__file__), sys.stdin.read(), sys.argv[1:]).run())
```

Comment:
Regarding python best practices for a script I would add `if __name__ == "__main__":` for this line.

And also used named arguments to be more readable like `info(spacing=10, collapse=1):`